### PR TITLE
Smith work, Atom creation

### PR DESCRIPTION
**Pull request by Stubeans**
Atom syntax code

### CODE REVIEW

File: src/Adder/Lang/Lexer.hs [hunk 1]
```diff
@@ -34,7 +34,7 @@ adderLexSpec =
       -- TODO Define the reserved names/keywords for the Adder language
       -- See https://docs.python.org/3/reference/lexical_analysis.html#keywords
       Tok.reservedNames =
-        [],
+        ["yield", "from"],
```
Comment (Tientuine) [inline src/Adder/Lang/Lexer.hs:37]: Fine to have these here, but we will probably not implement these kinds of statements in the interpreter.

**Review by Tientuine — CHANGES_REQUESTED** [review]
We discussed in person in class, so I won't say much more here unless you have new questions that come up.

File: src/Adder/Lang/Lexer.hs [hunk 2]
```diff
@@ -34,7 +34,7 @@ adderLexSpec =
       -- TODO Define the reserved names/keywords for the Adder language
       -- See https://docs.python.org/3/reference/lexical_analysis.html#keywords
       Tok.reservedNames =
-        ["or", "and", "is", "not", "in", "not in", "is not"],
+        ["yield", "from", "or", "and", "is", "not", "in", "not in", "is not"],
```
Comment (Tientuine) [inline src/Adder/Lang/Lexer.hs:37]: Let's not worry about `yield`, that's one of the language features that we said we would *not* try to implement.

File: app/Main.hs [hunk 3]
```diff
@@ -7,6 +7,7 @@
  -  This module provides routines for executables based on HOPL languages.
  -
  -  Author: Matthew A Johnson
+ -  Test Line
```
Comment (Tientuine) [inline app/Main.hs:10]: Get rid of this test line, only useful changes should be included!

File: src/Adder/Defs.hs [hunk 4]
```diff
@@ -19,4 +19,4 @@ type Source = String
 type Reference = Int
 
 -- Interpreter takes source code and produces a value (of some type)
-type Interpreter a = Source -> Either ParseError a
+type Interpreter a = Source -> Either ParseError a
```
Comment (Tientuine) [inline src/Adder/Defs.hs:22]: Undo your change to this file - somehow you removed the trailing newline, and there's no need for that change.

File: src/Adder/Lang/Parser.hs [hunk 5]
```diff
@@ -108,12 +108,56 @@ table =
     [Infix (reserved "or" >> return (BinaryExpr Or)) AssocLeft]
   ]
 
+atom :: IParser Expression
+atom = AtomExp <$> Id
```
Comment (Tientuine) [inline src/Adder/Lang/Parser.hs:112]: This is incorrect, but I believe we already discussed this and you have an un-pushed correction for this on your computer.

**Review by Tientuine — CHANGES_REQUESTED** [review]
Almost there... I believe you might already have a complete version that's waiting to be pushed, so I will keep an eye out for that.

**Comment by Tientuine** [comment]
@Stubeans This piece - identifiers - is one of the crucial pieces remaining in order for us to test parsing a variety of simple programs. Let's get it done! Please let me know if you need help to wrap this one up.

**Comment by Stubeans** [comment]
@Tientuine I believe I fixed the issues and fixed the Merge Conflicts correctly.

**Review by Tientuine — APPROVED** [review]

**Pull request merged**